pkg/internal/helper/helper.go

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *
+ */
+
+package helper
+
+import (
+	"fmt"
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt"
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/apis/kubevirt/validation"
+	"github.com/gardener/gardener/extensions/pkg/controller"
+	"github.com/gardener/gardener/extensions/pkg/controller/common"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func GetCloudProfileConfig(ctx *common.ClientContext, cluster *controller.Cluster) (*kubevirt.CloudProfileConfig, error) {
+	var cloudProfileConfig *kubevirt.CloudProfileConfig
+	if cluster != nil && cluster.CloudProfile != nil && cluster.CloudProfile.Spec.ProviderConfig != nil && cluster.CloudProfile.Spec.ProviderConfig.Raw != nil {
+		cloudProfileConfig = &kubevirt.CloudProfileConfig{}
+		if _, _, err := ctx.Decoder().Decode(cluster.CloudProfile.Spec.ProviderConfig.Raw, nil, cloudProfileConfig); err != nil {
+			return nil, errors.Wrapf(err, "could not decode providerConfig of cloudProfile for '%s'", cluster.Shoot.Name)
+		}
+		// TODO validate cloud profile on admission instead
+		if errs := validation.ValidateCloudProfileConfig(cloudProfileConfig); len(errs) > 0 {
+			return nil, errors.Wrap(errs.ToAggregate(), fmt.Sprintf("validation of providerConfig of cloud profile %q failed", cluster.CloudProfile.Name))
+		}
+	}
+	return cloudProfileConfig, nil
+}
+
+func GetControlPlaneConfig(ctx *common.ClientContext, cluster *controller.Cluster) (*kubevirt.ControlPlaneConfig, error) {
```

Comment:
I think that's not used

Reply:
None of the functions in this file are used, see my comment above.

Reply:
removed